Source/Scene/PostProcessorScene.js

```diff
@@ -20,23 +20,247 @@ define([
+
+
+            'bool bSpace = true; \n' +
+
+            //whether it is in space or not
+            'if(length(czm_viewerPositionWC.xyz) < 6500000.0) \n' +
+            '   bSpace = false; \n' +
+
+            //Sun position
+            'vec4 SunPos; \n' +
+            'if (czm_morphTime == 1.0) \n' +
+            '{                         \n' +
+            '    SunPos = vec4(czm_sunPositionWC, 1.0);   \n' +
+            '}                                              \n' +
+            'else \n' +
+            '{ \n' +
+            '    SunPos = vec4(czm_sunPositionColumbusView.zxy, 1.0); \n' +
+            '} \n' +
+
+            'vec4 SunPositionEC = czm_view * SunPos; \n' +
+            'vec4 SunPositionWC = czm_eyeToWindowCoordinates(SunPositionEC); \n' +
+
+
+            'SunPos = czm_viewportOrthographic * vec4(SunPositionWC.xy, -SunPositionWC.z, 1.0); \n' +
+            'vec2 texcoord = -v_textureCoordinates + vec2(1.0);   \n' +
+
+            'vec2 texelSize = 1.0 / czm_viewport.zw;    \n' +
+            'vec3 distortion = vec3(-texelSize.x * u_Distortion, 0.0, texelSize.x * u_Distortion);    \n' +
+
+            // ghost vector to image centre:
+            'vec2 ghostVec = (vec2(0.5) - texcoord) * u_GhostDispersal;  \n' +
+            'vec3 direction = normalize(vec3(ghostVec, 0.0));  \n' +
+
+            // sample ghosts:  
+           ' vec4 result = vec4(0.0);  \n' +
+           ' vec4 ghost = vec4(0.0);  \n' +
+           ' for (int i = 0; i < 4; ++i) {  \n' +
+           '     vec2 offset = fract(texcoord + ghostVec * float(i));  \n' +
+
+                //Only bright spots from the centre of the source image
+           //'     float weight = length(vec2(0.5) - offset) / length(vec2(0.5));  \n' +
+           //'     weight = pow(1.0 - clamp(weight,0.0,1.0) , 10.0);  \n' +
+
+            '    ghost += textureDistorted(u_colorTexture, offset, direction.xy, distortion, bSpace);  \n' +
+
+            '}  \n' +
+
+            ' result += ghost;  \n' +
+
+            // sample halo:
+            'vec4 halo;  \n' +
+            ' vec2 haloVec = normalize(ghostVec) * u_HaloWidth;  \n' +
+            ' float WeightForHalo = length(vec2(0.5) - fract(texcoord + haloVec)) / length(vec2(0.5));  \n' +
+            ' WeightForHalo = pow(1.0 - WeightForHalo, 5.0);  \n' +
+            ' halo = textureDistorted(u_colorTexture, texcoord + haloVec, direction.xy, distortion, bSpace) * WeightForHalo * 1.5;  \n' +
+
+            ' result += halo; \n' +
+            ' result += texture2D(u_DirtTexture, v_textureCoordinates);  \n' +
+
+          
+
+
+            //Rotating starburst texture's coordinate
+            ' vec3 camx = vec3(czm_view[0][0], czm_view[0][1], czm_view[0][2] ); \n' +
+            ' vec3 camz = vec3(czm_view[1][0], czm_view[1][1], czm_view[1][2] );  \n' +
+            ' float camrot = dot(camx, vec3(0, 0, 1)) + dot(camz, vec3(0, 1, 0));  \n' +
+
+            ' mat3 rotation = mat3(  \n' +
```

Comment:
I don't trust that all GLSL compilers for factor out the common `cos` and `sin` calls.  Call each function once above here and assign to local variables.